pkg/scd/operational_intents_handler.go

```diff
@@ -692,6 +742,14 @@ func (a *Server) upsertOperationalIntentReference(ctx context.Context, authorize
 			return stacktrace.Propagate(err, "Failed to upsert OperationalIntent in repo")
 		}
 
+		// Check if the previously attached subscription should be removed
```

Comment:
My reply to a comment on a [previous revision](https://github.com/interuss/dss/pull/1057#discussion_r1728517254) still applies:

> At this point in the transaction the subscription cannot be deleted, because it is referenced by the operational intent.
> 
> The previous implicit subscription can only be removed once the OIR has been updated to point to another subscription.
>
> If you delete it before the r.UpsertOperationalIntent(ctx, op) statement at line 719, you get an error.

Moving the deletion to just before `r.UpsertOperationalIntent(ctx, op)` at line 740 will cause the scenario that checks implicit subscription handling to fail (I just checked again to make sure).

Reply:
My bad, OK I see!